docs/book/src/clusterctl/provider-contract.md

```diff
@@ -241,6 +245,40 @@ Templates writers should use the common variables to ensure consistency across p
 Additionally, the value of the command argument to `clusterctl generate cluster <cluster-name>` (`<cluster-name>` in this case), will
 be applied to every occurrence of the `${ CLUSTER_NAME }` variable.
 
+### ClusterClass definitions
+
+An infrastructure provider could publish a **cluster class definition** file to be used by `clusterctl generate cluster` that will be used along
+with the workload cluster templates.
+This is a single YAML with _all_ the objects required that make up the cluster class.
+
+The following rules apply:
+
+#### Naming conventions
+
+ClusterClass definitions MUST be stored in the same folder as the component YAML and follow this naming convention:
+1. The cluster class definition should be names `clusterclass-{class-name}.yaml`, e.g `clusterclass-prod.yaml`
+
+`{class-name}` is the name of ClusterClass. 
+
+Each provider should create user facing documentation with the list of available cluster class definitions.
+
+#### Target namespace
+
+The cluster class definition YAML MUST asume the target namespace already exists.
+
+All objects in the cluster class definition YAML MUST be deploymned in the same namespace.
+
+It is strongly recommended that none of the objects in the cluster class definitions are shared acorss multiple definitions.
```

Comment:
```suggestion
It is strongly recommended that none of the objects in the cluster class definitions are shared across multiple definitions.
```
nit

Reply:
```suggestion
Even if technically possible, it is strongly recommended that none of the objects in the ClusterClass definitions are shared acorss multiple definitions; this helps on preventing that changing on an object impacts inadvertently many ClusterClasses, and consequently, all the Clusters using those.
```

Reply:
If this something we want to prevent from happening and validate with webhooks?

Reply:
> If this something we want to prevent from happening and validate with webhooks?

Advanced users can choose to opt-in and reusing templates, it could be a powerful feature in the proper hands.
However, but given that in this context are talking about template for the quick start UX let reccomand the safest path.

Reply:
>If this something we want to prevent from happening and validate with webhooks?

We would not want to prevent this as a feature in ClusterClass. As @fabriziopandini mentioned reusing templates can become a really powerful feature that advanced users will want to use. 

Given that this doc mostly focuses on 'example' ClusterClass definitions that a provider can provide, which in most cases will be used for quick start, we would want to recommend the safest path.